code/datums/components/cleaning.dm

```diff
@@ -0,0 +1,38 @@
+/datum/component/cleaning
+
+/datum/component/cleaning/Initialize()
+	if(!ismovable(parent))
+		. = COMPONENT_INCOMPATIBLE
+		CRASH("[type] added to a [parent.type]")
+	RegisterSignal(list(COMSIG_MOVABLE_MOVED), .proc/Clean)
+
+/datum/component/cleaning/proc/Clean()
+	var/turf/tile = AM.loc
+	if(!isturf(tile))
+		return
+
+	tile.clean_blood()
+	for(var/A in tile)
```

Comment:
This entire block of code seems cancerous and like it would cause lag, Correct me if i'm wrong.

Reply:
Recently cleaning related code has been getting better but this can't be removed quite yet. We already use this exact code.

Lag wise though it's not too terrible.